inference-engine/src/plaidml_plugin/ops/shuffle_channels.cpp

```diff
@@ -17,28 +17,26 @@ static OpRegistration reg("shuffleChannels", [](const Context& ctx) {
   IE_ASSERT(ctx.operands.size() == 1);
   auto I = ctx.operands.at(0);
   auto group = layer->get_group();
-  //  the axis parameter doesn't show in the formula.
-  //  not sure why we need axis, so just leave it alone for now.
+  auto axis = layer->get_axis();
+  if (axis < 0) {
+    axis = axis + 4;
```

Comment:
Change 4 to `I.rank()`. Perhaps add `IE_ASSERT(I.rank() == 4)` since this is by spec a rank 4 tensor. I think this makes it clearer what this code is doing.